### PR TITLE
Feature/fetch plots

### PR DESCRIPTION
**Pull request by masafumimori**
In this PR
- modified Prettier settings
- add API mock to return mocked TRDL data
- draw map based on data returned from API
- show data returned from API in modal

I have some concerns:
1. Should I use `plot` instead of `tile` for the type and variable names? (now they are mixed so being consistent would be good)
2. I am not sure if the directory structure is right. For example, I added `mock-data.ts` under `pages/api/nfts/type` but should I create another folder containing only mock data?
3. Is it ok to have constants in helper files? I know that sometimes constants are written in the `constant` files/folders so I would like to check your preference about it.

### CODE REVIEW

File: .prettierrc.json [hunk 1]
```diff
@@ -2,7 +2,8 @@
     "semi": false,
     "tabWidth": 4,
     "singleQuote": true,
-    "printWidth": 80,
+    "printWidth": 120,
     "trailingComma": "none",
-    "bracketSameLine": true
+    "bracketSameLine": true,
+    "arrowParens": "avoid"
```
Comment (lorenzopolidori) [inline .prettierrc.json:8]: 👍

File: components/map/index.tsx [hunk 2]
```diff
@@ -31,21 +32,10 @@ const HORIZONTAL_SCROLL_SENSITIVITY = 0.05
...
-    // This shows which tile image should be displayed(index of TILE_TEXTURES fetched by getTileImages())
-    const tileMap = [
-        14, 23, 23, 23, 23, 23, 23, 23, 23, 13, 21, 32, 33, 33, 28, 33, 33, 33,
-        31, 20, 21, 34, 9, 9, 34, 1, 1, 1, 34, 20, 21, 34, 4, 4, 34, 1, 1, 10,
-        34, 20, 21, 25, 33, 33, 24, 33, 33, 33, 27, 20, 21, 34, 4, 7, 34, 18,
-        17, 10, 34, 20, 21, 34, 6, 8, 34, 16, 19, 10, 34, 20, 21, 34, 1, 1, 34,
-        10, 10, 10, 34, 20, 21, 29, 33, 33, 26, 33, 33, 33, 30, 20, 11, 22, 22,
-        22, 22, 22, 22, 22, 22, 12
-    ]
+    const [tileMaps, setTileMaps] = useState<TileMapType[]>([])
```
Comment (lorenzopolidori) [inline components/map/index.tsx:38]: Maybe better `const [mapTiles, setMapTiles] = useState<MapTileType[]>([])`

File: components/map/index.tsx [hunk 3]
```diff
@@ -65,14 +55,16 @@ const Map = ({ width, height, headerHeight, onSelectTile }: MapProps) => {
 
     const renderTiles =
         (ctx: CanvasRenderingContext2D) => (x: number, y: number) => {
-            const images = getTileImages()
+            if (tileMaps.length === 0) return
 
             for (let tileX = 0; tileX < GRID_SIZE; ++tileX) {
                 for (let tileY = 0; tileY < GRID_SIZE; ++tileY) {
-                    const imageIndex = tileMap[tileY * GRID_SIZE + tileX]
+                    const index = (tileY * GRID_SIZE + tileX) % 14
```
Comment (lorenzopolidori) [inline components/map/index.tsx:62]: Why % 14?

File: utils/api-config.js [hunk 4]
```diff
@@ -14,13 +14,15 @@ const terracellContract = (id, applicationId) => `/api/terracells/${id}/contract
 const nfts = accountId => `/api/accounts/${accountId}/nfts`
 const fileUpload = '/api/files/upload'
 const ipfsFiles = '/api/ipfs/files'
+const plots = (symbol, next) => `/api/nfts/type/${symbol}${next ? `?next=${next}` : ''}`
```
Comment (lorenzopolidori) [inline utils/api-config.js:17]: This should be:
```js
const plots = next => `/api/nfts/type/trld${next ? `?next=${next}` : ''}`
```
as plots already implies that we are fetching `TRLD` NFTs.

Even better, I suggest we get more explicit in what we are fetching, as in:
```js
const terralands = next => `/api/nfts/type/trld${next ? `?next=${next}` : ''}`
```

File: pages/api/nfts/type/mock-data.ts [hunk 5]
```diff
@@ -0,0 +1,87 @@
+const assetList = [
```
Comment (lorenzopolidori) [inline pages/api/nfts/type/mock-data.ts:1]: 👍

File: pages/api/nfts/type/mock-data.ts [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+    {
+        id: '101844947',
+        name: 'Moorland#1@arc3',
+        symbol: 'TRLD',
+        url: 'ipfs://QmecxV7vrVEUTjvtu7VutqxhoYLFGtEENXvhBPAcJLSHxH',
+        offchainUrl: 'https://images.terragrids.org/map-tile-grass.png',
+        position: '0,0',
```
Comment (lorenzopolidori) [inline pages/api/nfts/type/mock-data.ts:8]: Let's change this to:
```
...
positionX: 0,
positionY: 0,
...
```
as this is how I ended up implementing the real API: https://github.com/terragrids/api/blob/dev/src/repository/token.repository.js#L77

**Comment by lorenzopolidori** [comment]
I have noticed that when I click on a plot, the information does not refer to the plot I selected, see video attached.

https://user-images.githubusercontent.com/2437709/182816688-9e84f05f-79bf-44cf-a790-55ab0ab196d5.mov

File: components/map/index.tsx (hunk 3 above)
Comment (masafumimori) [reply components/map/index.tsx]: Sorry I forgot to delete this. I was using this when I was trying to see the API response works ( at the time only 15 elements existed)

**Comment by masafumimori** [comment]
@lorenzopolidori 
Thank you for reviewing and answering my questions.

I fixed where was pointed out.

>I have noticed that when I click on a plot, the information does not refer to the plot I selected, see video attached.
For this, I think it was due to the code I forgot to delete, `% 14` part, as it limits to use only elements between index 0 ~ 14 so there should have been some mismatch between actual `plotMaps` data and rendered image. (within `onClick` function, index wasn't calculated with remainder (%14), so true `plotMaps` data was shown in modal)
I deleted `%14` and it seems working fine.

**Review by lorenzopolidori — APPROVED** [review]

**Pull request merged**